.github/workflows/packages.yml

```diff
@@ -0,0 +1,105 @@
+      - 'include/**'
+      - 'package/**'
+      - 'toolchain/**'
+      - 'config/**'
+jobs:
+  build:
+    name: Build packages with external toolchain
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: False
+      matrix:
+        include:
+          - name: malta-be
+            target: malta
+            subtarget: be
+          - name: x86-64
+            target: x86
+            subtarget: 64
+
+    container: registry.gitlab.com/openwrt/buildbot/buildworker-3.4.1
+
+    steps:
+      - name: Checkout master directory
+        uses: actions/checkout@v2
+        with:
+          path: openwrt
+
+      - name: Fix permission
+        run: |
+          chown -R buildbot:buildbot openwrt
+
+      - name: Initialization environment
+        run: |
+          TARGET=$(echo ${{ matrix.target }})
+          SUBTARGET=$(echo ${{ matrix.subtarget }})
+          echo "TARGET=$TARGET" >> "$GITHUB_ENV"
+          echo "SUBTARGET=$SUBTARGET" >> "$GITHUB_ENV"
+
+      - name: Parse toolchain file
+        working-directory: openwrt
+        run: |
+          TOOLCHAIN_FILE=$(curl "https://downloads.cdn.openwrt.org/snapshots/targets/${{ env.TARGET }}/${{ env.SUBTARGET }}/sha256sums" \
+            | sed -n -e 's/.*\(openwrt-toolchain.*\).tar.xz/\1/p')
+
+          echo "TOOLCHAIN_FILE=$TOOLCHAIN_FILE" >> "$GITHUB_ENV"
+
+      - name: Download external toolchain
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: |
+          wget -O - https://downloads.cdn.openwrt.org/snapshots/targets/${{ env.TARGET }}/${{ env.SUBTARGET }}/${TOOLCHAIN_FILE}.tar.xz \
+            | tar --xz -xf -
+
+      - name: Create configuration
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: |
+          echo CONFIG_ALL=y >> .config
+          echo CONFIG_ALL_KMODS=y >> .config
+          echo CONFIG_ALL_NONSHARED=y >> .config
+
+          ./scripts/ext-toolchain.sh \
+            --toolchain ${{ env.TOOLCHAIN_FILE }}/toolchain-* \
+            --overwrite-config \
+            --config ${{ env.TARGET }}/${{ env.SUBTARGET }}
+
+          make defconfig
+
+      - name: Show configuration
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: ./scripts/diffconfig.sh
+
+      - name: Build tools
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: make tools/install -j$(nproc) BUILD_LOG=1
+
+      - name: Build toolchain
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: make toolchain/install -j$(nproc) BUILD_LOG=1
+
+      - name: Build Kernel
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: make target/compile -j$(nproc) BUILD_LOG=1
+
+      - name: Build everything
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
+        run: make -j$(nproc) BUILD_LOG=1
```

Comment:
I think we have to build everything separately. A simple make won't permit us to skip things like kernel build

Reply:
In that case make package/compile should be used. Running each individually would take forever as it checks the dependencies every time

Reply:
Yes I mean using all the subcommand, wasn't saying to call compile for each package.

Reply:
I would like to compile all normal and kernel packages from the OpenWrt core here explicitly.
I restricted it to only two targets, but on these I would like to run a full build to hopefully spot build problems early.

Reply:
@hauke to make debug quicker i notice a little trick on another workflow.

`make -j$(nproc) BUILD_LOG=1 || make -j1 V=s` in theory the affected error should be printed. What do you think?

Reply:
> make -j1 V=s

You should probably first backup the logs which would be likely containing the previous error(s), since it might be related to some concurrent build issue and thus single threaded build attempt might succeed and clear out that previous build error(s).

>  in theory the affected error should be printed

What about following [ci: show build failures directly in job log output](https://github.com/ynezz/openwrt/commit/2dc201b4d40fdb8217522a3757638573589cfbc3) ?

Reply:
I copied it tform https://github.com/ynezz/openwrt/commit/2dc201b4d40fdb8217522a3757638573589cfbc3